### PR TITLE
Add camera

### PR DESCRIPTION
**Pull request by ilcheese2**

### CODE REVIEW

File: pyweek36/main.py [hunk 1]
```diff
@@ -195,9 +199,17 @@ def on_update(self, delta_time):
         # Move items in the physics engine
         self.physics_engine.step()
 
+        self.camera.move_to(
```
Comment (n0Oo0Oo0b) [inline pyweek36/main.py:202]: https://github.com/letsbuilda/pyweek36/blob/main/pyweek36/main.py#L155-L156

Moving the camera will offset the cursor position, so you'll have to adjust for that in the bullet angle calculation too

File: pyweek36/main.py [hunk 2]
```diff
@@ -195,9 +199,17 @@ def on_update(self, delta_time):
...
+            (
+                max(self.player_sprite.center_x - self.camera.viewport_width / 2, 0),
+                max(self.player_sprite.center_y - self.camera.viewport_height / 2, 0),
```
Comment (n0Oo0Oo0b) [inline pyweek36/main.py:205]: The current camera movement is quite sharp, I think adding some damping could help

We could also add a deadzone so the camera doesn't move every time you move/jump, and lookahead on the x axis

**Review by n0Oo0Oo0b — CHANGES_REQUESTED** [review]

File: pyweek36/constants.py [hunk 3]
```diff
@@ -67,3 +67,5 @@ class InputType(IntEnum):
 DARKMATTER_TEXTURE = arcade.load_texture(
     ASSETS_DIR / "sprites" / "map" / "dark_matter.png"
 )
+
+CAMERA_DAMPING = 0.8
```
Comment (n0Oo0Oo0b) [inline pyweek36/constants.py:71]: I think 0.8 is quite hard to notice, from my testing somewhere around 0.15-0.3 feels a lot more natural

**Review by n0Oo0Oo0b — DISMISSED** [review]
Looks alright to me, but another review from someone else won't hurt since I made some changes myself

**Comment by Anonymous4045** [comment]
There's some conflicts

**Comment by n0Oo0Oo0b** [comment]
> There's some conflicts

Resolved

**Review by n0Oo0Oo0b — APPROVED** [review]
LGTM I guess

**Pull request merged**